core/src/database/DatabaseSessionPool.cpp

```diff
@@ -88,7 +92,7 @@ namespace ledger {
             int version = getDatabaseMigrationVersion(sql);
 
             soci::transaction tr(sql);
-            migrate<CURRENT_DATABASE_SCHEME_VERSION>(sql, version);
+            migrate<CURRENT_DATABASE_SCHEME_VERSION>(sql, version, _usingPostgreSQL);
```

Comment:
As @alekece said, I also think we need a better representation of backends. Booleans don’t fit well here (booleans actually never fit well anything most of the type; it’s better to use typed enums).

Reply:
I wanted to be part of a more general conversation, but I introduced a new enum `DatabaseBackendType` to handle that !